tests/components/element-editor/basic-details/basic-details.spec.tsx

```diff
@@ -77,7 +77,7 @@ describe("BasicDetails", () => {
         });
 
         it("should replace element name, mandatory field's error message, and ID with values that correspond to the new element type", async () => {
```

Comment:
actually the fields got updated / dirtied because of the useEffect in `TestHelper` so it is not an effective test.

you can explore mocking the `showModal` and calling the callbacks from there

```
const mockHideModal = jest.fn();
const mockShowModal = jest.fn();
jest.mock("src/context-providers/display/modal-hook", () => {
	return {
		useModal: () => ({
			hideModal: mockHideModal,
			showModal: mockShowModal,
		}),
	};
});
// ...

it("should replace element name, mandatory field's error message, and ID with values that correspond to the new element type", async () => {
  // trigger reset confirmation modal
  expect(mockShowModal).toHaveBeenCalled();
  mockShowModal.mock.lastCall[1].onClickActionButton(); // <- this triggers the reset
});
```

Reply:
Thank you for this, did not know it was possible to trigger the actual callbacks.